pkg/controller/attach_detach.go

```diff
@@ -50,6 +52,9 @@ import (
 type CloudDiskAttachDetacher interface {
 	PublishVolume(ctx context.Context, volumeID string, nodeID string, volumeContext map[string]string) provisioner.CloudAttachResult
 	UnpublishVolume(ctx context.Context, volumeID string, nodeID string) error
+	GetNodeDataDisks(nodeName types.NodeName, crt azcache.AzureCacheReadType) ([]compute.DataDisk, *string, error)
```

Comment:
Let's avoid exposing the `cloud-provider-azure` types via this interface and drop the `crt` parameter. I doubt that we would ever need to use anything by the default.

Reply:
I'm wondering what's the reason that we need to avoid exposing `types.NodeName`?

Reply:
I was referring to `azcache.AzureCacheReadType`. `types.NodeName` is a Kubernetes type.

Reply:
Done.